### ISSUE TITLE
Can't login with anilist

### ISSUE THREAD

**Issue by lewdpatriot** [ThaUnknown/miru#153]
It redirects to firefox and fails to load the page. 
![IMG_20220418_212814](https://user-images.githubusercontent.com/63286528/163836513-3caf0278-d151-4777-94d1-a06e3b0be54b.png)

**Comment by ThaUnknown** [comment]
firefox issue lol, open it with chrome i guess, there's no way for me to fix a firefox bug

**Comment by lewdpatriot** [comment]
It's showing this in brave and chromium.
![IMG_20220418_221645](https://user-images.githubusercontent.com/63286528/163842357-ba90b06a-1a7c-4028-ac30-c135b29659d7.png)
If I click open xdg-open this happens, if not, blank page. Am I missing something? 
![IMG_20220418_221608](https://user-images.githubusercontent.com/63286528/163842405-82323bf4-8fd2-4c13-a5c4-f7ac7142277c.png)

**Comment by ThaUnknown** [comment]
very smartn't of you to post your token in here, it seems to be an issue with linux, i dont know how to fix it

**Comment by lewdpatriot** [comment]
Oops. Ok, thanks. Aside from this issue it's a wonderful app

**Comment by Muril-o** [comment]
Probably .AppImage but just to confirm since you didn't bother filling out the issue template I'll ask:
Is this .AppImage or .deb?

**Comment by lewdpatriot** [comment]
> Probably .AppImage but just to confirm since you didn't bother filling out the issue template I'll ask: Is this .AppImage or .deb?

It is appimage. Now I see the issue. I don't really use appimage, but since I'm on arch had no other choice. Is there a solution?

**Comment by Muril-o** [comment]
Since appimages are standalone-ish youll need to manually mime the protocol to your user
tl;dr; create a .desktop for miru first, depending on your distro youll need to symlink from local to usr and then xdg-mime it to the .desktop

`choose a directory for the application, move the appimage to the directory and cd into it`
then
`$: ./Miru.AppImage --appimage-extract`
this will extract stuff to a squashfs-root, inside there will be a miru.desktop
`edit it to point 'Exec=' to the appimage location (use the complete path)`
then copy it to local apps
`$: cp miru.desktop ~/.local/share/applications/`
you can symlink to usr if necessary (depends on your distro)
`$: ln -s ~/.local/share/applications/miru.desktop /usr/share/applications/miru.desktop`
then xdg-mime to default the protocol to the .desktop
`$: xdg-mime default miru.desktop x-scheme-handler/miru`
you might wanna do application too
`$: xdg-mime default miru.desktop application/miru`

**Comment by lewdpatriot** [comment]
> Since appimages are standalone-ish youll need to manually mime the protocol to your user tl;dr; create a .desktop for miru first, depending on your distro youll need to symlink from local to usr and then xdg-mime it to the .desktop
> 
> `choose a directory for the application, move the appimage to the directory and cd into it` then `$: ./Miru.AppImage --appimage-extract` this will extract stuff to a squashfs-root, inside there will be a miru.desktop `edit it to point 'Exec=' to the appimage location (use the complete path) For example: Exec=/home/your_user/Downloads/miru/squashfs-root/Miru.AppImage` then copy it to local apps `$: cp miru.desktop ~/.local/share/applications/` you can symlink to usr if necessary (depends on your distro) `$: ln -s ~/.local/share/applications/miru.desktop /usr/share/applications/miru.desktop` then xdg-mime to default the protocol to the .desktop `$: xdg-mime default miru.desktop x-scheme-handler/miru` you might wanna do application too `$: xdg-mime default miru.desktop application/miru`

It worked. Yayy! Thank you a lot

**Comment by lewdpatriot** [comment]
@ThaUnknown it's not the app's bug. As appimages are isolated it needs a bit tinkering to integrate it to the system. I didn't know how to do it, so it's completely my fault. You can close this

**Comment by ThaUnknown** [comment]
> @ThaUnknown it's not the app's bug. As appimages are isolated it needs a bit tinkering to integrate it to the system. I didn't know how to do it, so it's completely my fault. You can close this

![image](https://user-images.githubusercontent.com/6506529/164224268-e3a259b3-4ede-46bb-8826-a3ae19c1bf64.png)
you will get this if you're on linux now, and you can just paste the URL with the token into the app

**Comment by YoruNoKen** [comment]
Hello, didn't know if I should've opened a new issue or reopened this one,
![image](https://github.com/ThaUnknown/miru/assets/69751661/3476d018-c780-43c5-b031-e414e882eeac)
After I've applied the fix, it's prompting with this error.
